src/minter-contract-utils/src/evm_bridge.rs

```diff
@@ -108,6 +111,37 @@ impl EvmParams {
             next_block: next_block.0.as_u64(),
             nonce: nonce.0.as_u64(),
             gas_price,
+            last_log_block: None,
         })
     }
+
+    /// Gets the last log block to fetch logs from.
+    ///
+    /// If `None`, fetch until `Safe`
+    pub fn last_log_block(&self) -> BlockNumber {
+        self.last_log_block
+            .map(|block| BlockNumber::Number((self.next_block + block).into()))
+            .unwrap_or(BlockNumber::Safe)
+    }
+
+    /// Set last log block to fetch after an error fetching logs.
+    ///
+    /// If last_log_block is None, it will be set to 128.
+    /// If last_log_block is 1, it will be set to 1 and the next block will be incremented.
+    /// Otherwise, it will be set to half of the current value.
+    pub fn set_last_log_block_error(&mut self) {
```

Comment:
This logic is confusing and does not fix what reported by F11. The issue has to be fixed by paginating if more than X blocks are to be queried